azurerm/internal/services/web/data_source_function_app_host_keys.go

```diff
@@ -0,0 +1,86 @@
+package web
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmFunctionAppHostKeys() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmFunctionAppHostKeysRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"master_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"function_keys": {
+				Type:     schema.TypeMap,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"system_keys": {
```

Comment:
I am not sure about system_keys .. it exists in the api, however I never used it nor was able to see it being set while I was testing

Reply:
We can simply just keep it here as long as the test passe :+1:

Reply:
If its value is not returned by the API / we're not using it, then it should be removed.

Reply:
I agree with @jackofallops, I am going to remove it